app/controllers/admins_controller.rb

```diff
@@ -0,0 +1,36 @@
+ 
+class AdminsController < ApplicationController
+    #prepend_before_action :require_no_authentication, only: [:cancel]
+    before_action :authorize_admin
+    # skip_before_action :verify_signed_out_user 
+    
+    def index
+        @users = User.all
+    end
+    
+    def edit
+        @user = User.find(params[:id])
+    end
+    
+    def update 
+        #This gets the info from the edit form and updates the data. 
+        @user = User.find(params[:id])
+        @user.email = params[:user][:email]
+        @user.password = params[:user][:password]
+        @user.password_confirmation = params[:user][:password_confirmation]
+        @user.first_name = params[:user][:first_name]
+        @user.last_name = params[:user][:last_name]
+        @user.phone = params[:user][:phone]
```

Comment:
Also, you don't really need an instance variable, e.g. `@user` .   You can just use user as the variable name, because you aren't passing it to any erb file

Reply:
I've taken care of this as well.

Reply:
Leila, I hadn't seen your comment when I said that. I can put it back as an instance variable again.